src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -1626,3 +1626,42 @@ def test_siblings(fortran_reader):
     root_siblings = psyir.siblings
     assert len(root_siblings) == 1
     assert root_siblings[0] is psyir
+
+
+def test_walk_depth(fortran_reader):
+    '''Test the depth restriction functionality of Node's walk method.'''
+
+    code = '''subroutine test_depth()
+    integer :: i, j, k
+    integer :: arr(2,2,2)
+
+    do i = 1, 2
+      do j = 1, 2
+        do k = 1, 2
+          arr(i,j,k) = 0
```

Comment:
For my peace of mind, would you mind increasing the complexity of this a bit so that we exercise going down different branches of the tree. e.g. change the innermost loop body to be:

    if(something)then
      arr(i,j,k) = 0
    else
      arr(i,j,k) = -1
    end if

so that we have two assignments at the same depth but in different branches.

Reply:
Good idea. Addressed in 9b7917c75.